File: src/BankAccount.py
from clearScreen import clearScreen


class BankAccount:

    def __init__(self, initialDeposit, password=None):
        self.balance = float(initialDeposit)
        if password is None:
            self.SetPassword()

    def SetPassword(self):
        global password
        password = input('Please enter a password for your account: ')
        if password == '':
            print("Passwords cannot be empty!")
            clearScreen()
            self.SetPassword()
        else:
            if password is not None:
                confirmPassword = input('Please input your password one more time to confirm it! ')
                if password != confirmPassword:
                    print('Your passwords to not match... ')
                    self.SetPassword()
                else:
                    print('Password set! Your account is now ready! ')
                    return
            else:
                print("Please enter a password! ")
                self.SetPassword()

    # Returns true if you have more balance than cost, false if you don't
    def IsItemAffordable(self, amount):
        if float(amount) <= self.balance:
            return True
        else:
            return False

    def MakePurchase(self, amount):
        if self.VerifyPassword():
            if amount <= self.balance:
                self.balance -= amount
                print(f'{amount} spent from your account.')
                print(f'You now have ${self.balance} remaining.')
                return True
            else:
                print('You do not have enough funds left to afford this item.')
                return False
        else:
            return False

    def MakeDeposit(self, depositAmount):
        if self.VerifyPassword():
            self.balance += depositAmount
            print(f"Deposit completed! you now have ${self.balance} on your account!")

    def MakeInitialDeposit(self, initialDepositAmount):
        self.balance += initialDepositAmount

    def BalanceReport(self):
        print(f'You have $ {self.balance} left in your account.')

    def VerifyPassword(self):
        passEntry = input('Please enter your password to confirm your identity: ')
        if passEntry == password:
            return True
        else:
            print('Incorrect password!')
            return False

File: src/main/java/BankAccount.java

import java.util.Scanner;

public class BankAccount
{
    
    Scanner scan = new Scanner(System.in);
    
    private String password;
    private double balance;

    public BankAccount(double initialDeposit)
    {
        balance = initialDeposit;
        setPassword();
    }
    
    // Ask the user to set password, requiring two successful entries to lock it in permanently
    private void setPassword()
    {
        System.out.println("Please enter a password for your account: ");
        password = scan.nextLine();
        System.out.println("Confirm your password by entering it one more time!");
        String tempPassCheck = scan.nextLine();
        if(!password.equals(tempPassCheck))
        {
            System.out.println("Your passwords do not match ... ");
            setPassword();
        }
        else
        {
            System.out.println("Password set! Your account is now ready with a balance of " + balance);
        }
    }
    
    public boolean canAfford(double amount)
    {
        if(amount <= balance)
        {
            return true;
        }
        else
        {
            System.out.println("You don't have enough money in your account.");
            return false;
        }            
    }
    
    // Allow the user to attempt to withdraw money and report on success. If successful, adjust balance.
    public void makePurchase(double amount)
    {
        if(checkPassword())
        {
            if(amount <= balance)
            {
                balance -= (float) amount;
                System.out.println("" + amount + " spent from your account");
                System.out.println("You have " + balance + " remaining.");
            }
            else
            {
                System.out.println("Not enough remaining funds");
            }
        }
    }
    
    public void depositMoney(double amount)
    {
        if(checkPassword())
        {
            balance += (float) amount;
            System.out.println("You added " + amount + " to your account.");
            System.out.println("You now have $" + balance + " available.");            
        }
    }
    
    public double getBalance()
    {
        return balance;
    }
    public void addBalance(double balanceToAdd)
    {
        balance += balanceToAdd;
    }
    private boolean checkPassword()
    {
        System.out.println("Please enter your password to access account: ");
        String passEntry = scan.nextLine();
        if(passEntry.equals(password))
        {
            return true;
        }
        else
        {
            System.out.println("Incorrect password!");
            return false;
        }
    }

}
